lib/options.go

```diff
@@ -285,6 +285,9 @@ type Options struct {
 
 	// Redirect console logging to a file
 	ConsoleOutput null.String `json:"-" envconfig:"K6_CONSOLE_OUTPUT"`
+
+	// Plugins to load
+	Plugins []string `json:"plugins" envconfig:"K6_PLUGINS"`
```

Comment:
Still need to figure out how this `envconfig` flag works. I couldn't get it to read from the environment but I'm probably just missing something really basic.

Reply:
envconfig has issues (https://github.com/loadimpact/k6/issues/671) and we want to replace it, but the problem here is not related to them... The reason you can't read from the environment is that this reading happens after you've already initialized the plugins. The "consolidated" version of `lib.Options` is constructed here: https://github.com/andremedeiros/k6/blob/9dbc2f9ff3d7dbceeb6f06c58b81dcfda3fb3243/cmd/run.go#L169-L172

And, as you can see, this "consolidation" consists of combining the {option defaults + JSON config options + **the script exported `options`** + environment variables + CLI flags}, in increasing order of precedence ([more info](https://k6.io/docs/using-k6/options#using-options)). That chicken-and-egg situation is why you literally can't fix this here, given that plugins would have to be available when the init context of the script is executed to get its exported `options`.

Instead of `lib.Options`, you should have the plugin list in [`RuntimeOptions`](https://github.com/loadimpact/k6/blob/master/lib/runtime_options.go), which contain things that might affect even the initial script execution (and thus, the exported script `options`), like environment variables. They can also easily be moved earlier in the `cmd/run.go` execution as well: https://github.com/loadimpact/k6/blob/0e94653c9954c17b2b1a480b456e23ec120feff2/cmd/run.go#L123-L126